### PR TITLE
fixes #6: adds exception handling in the registerTo method 

### LINKED ISSUE #6: Improve exception handling in AsciidoctorExtensions.registerTo
If there's an exception during registration in [registerTo](https://github.com/asciidoctor/asciidoctorj-groovy-dsl/blob/master/src/main/groovy/org/asciidoctor/groovydsl/AsciidoctorExtensions.groovy#L47) method, the whole process aborts but the _REGISTERED_EXTENSIONS_ is not cleared.
This means that:
- System aborts with unfriendly message
- Correctly declared extensions after point of error are not registered. This behaviour is debatable, but since previous extensions are already registered I think it would be nice that the system skips the error and continues. Another option is to unregister any previous extension, but is that even possible?
- _REGISTERED_EXTENSIONS_ remains with previous values, so if you try to register another extension in another point of the program it fails because it tries ti register the previous one. That affects for example the unit tests, if one fails, all the ones after that fails because of a previous one.

I don't have clear answer for all points, but to begin with I think we should handle any exception to offer a better message, and in case of error clear the _REGISTERED_EXTENSIONS_ list.

### PR DESCRIPTION
**Pull request by abelsromero**
This PR includes some simple exception handling to:
. Ensure that the temporal extension registry ('REGISTERED_EXTENSIONS' in 'AsciidoctorExtensions') is cleared in case of error.
. Improve exception messages to make debugging easier

Note: clear() invocation must be placed in catch bocks instead of finally to prevent concurrent exception.

### CODE REVIEW

File: src/main/groovy/org/asciidoctor/groovydsl/AsciidoctorExtensions.groovy [hunk 1]
```diff
@@ -46,26 +46,39 @@ class AsciidoctorExtensions {
...
             switch (it) {
                 case Closure:
-                    it.delegate = extensionHandler
-                    it.call()
+                    try {
+                        it.delegate = extensionHandler
+                        it.call()
+                    } catch (e) {
+                        REGISTERED_EXTENSIONS.clear()
+                        throw new AsciidoctorExtentionException("Error registering extension from class in ${it.class.name}", e)
```
Comment (robertpanzer) [inline src/main/groovy/org/asciidoctor/groovydsl/AsciidoctorExtensions.groovy]: Codenarc doesn't like the GString here.
Could you change the `"Error..."` to `'Error...'`?

File: src/main/groovy/org/asciidoctor/groovydsl/AsciidoctorExtentionException.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package org.asciidoctor.groovydsl;
+
+
+/**
+ * Generic exception to manage Asciidoctor Extension DSL processing errors
+ */
+public class AsciidoctorExtentionException extends Exception {
```
Comment (robertpanzer) [inline src/main/groovy/org/asciidoctor/groovydsl/AsciidoctorExtentionException.java]: Could you rename this class to `AsciidoctorExtensionException`?

**Comment by robertpanzer** [comment]
Looks great so far!

**Comment by abelsromero** [comment]
Not having to squash speed things up  a lot :D

**Comment by abelsromero** [comment]
It's interesting that this also randomly failed for openjdk7...

**Comment by robertpanzer** [comment]
OpenJDK seems to be really messed up on Travis. Maybe we should simply deactivate it.

**Comment by robertpanzer** [comment]
Thanks for this!
This brings the required robustness to this project.

**Pull request merged**